common/ComputationalTools.h

```diff
@@ -0,0 +1,40 @@
+#ifndef COMPUTATIONAL_TOOLS
+#define COMPUTATIONAL_TOOLS
+
+template <typename T, typename Total, int N>
+class MovingAverage
+{
+	public:
+		MovingAverage()
+			: num_samples_(0), total_(0)
+			{ }
+
+	void Reset()
+	{
+		num_samples_ = 0;
+		total_ = 0;
+	}
+
+	void operator()(T sample)
+	{
+		if (num_samples_ < N)
```

Comment:
You could do it without the branch here. Just initialize the array to zero. It's also a little odd to overload operator() for this. I'd say make a separate method for taking a sample.